packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -274,6 +287,8 @@ class TextPainter {
   }
 
   /// The maximum width at which to layout the text.
+  ///
+  /// Requires [layout] or [layoutToMaxIntrinsicWidth] to be called again before painting.
```

Comment:
calling `layoutToMaxIntrinsicWidth` after setting this is pointless since it would overwrite it.